protocols/shell-workspace-creation-v1.xml

```diff
@@ -0,0 +1,321 @@
+      <description summary="destroy the manager">
+        Destroy the manager. Destroying the manager does not destroy objects
+        created by the manager.
+      </description>
+    </request>
+
+    <request name="get_relative_target">
+      <description summary="create a workspace target relative to another workspace">
+        This request creates a workspace target at the given coordinate offset
+        from another workspace. The workspace coordinate space should match the
+        one used in 'ext_workspace_handle_v1.coordinates'.
+
+        A given coordinate offset may be unrepresentable or otherwise disallowed
+        by the compositor, in which case the workspace target will be invalid.
+
+        If a workspace already exists at the given offset, the resuling
+        target should be valid and refer to the existing workspace.
+      </description>
+
+      <arg name="descriptor" type="new_id" interface="shell_workspace_creation_target_descriptor_v1"/>
+      <arg name="workspace" type="object" interface="ext_workspace_handle_v1"/>
+      <arg name="offset" type="array" summary="coordinate offset from the given workspace"/>
+    </request>
+  </interface>
+
+  <interface name="shell_workspace_creation_target_v1" version="1">
+    <request name="destroy" type="destructor">
+      <description summary="destroy the manager">
+        Destroy the manager. Destroying the manager does not destroy objects
+        created by the manager.
+      </description>
+    </request>
+
+    <description summary="a target for actions that may create or activate a target workspace">
+      A workspace creation target is a target for actions that may create a new workspace
+      if performed. It may refer to either an existing workspace, or act as a marker
+      of where and how a new workspace should be created.
+    </description>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the target">
+        Destroy the workspace target. Destroying the target does not
+        cancel committed requests.
+      </description>
+    </request>
+
+    <enum name="capabilities">
+      <entry name="create"
+             summary="the targeted workspace can be created with only a request"
+             value="1"/>
+      <entry name="external"
+             summary="the targeted workspace can be created through an extension protocol"
+             value="2"/>
+      <entry name="by_name"
+             summary="the targeted workspace can be created by assigning a name"
+             value="4"/>
+      <entry name="nameable"
+             summary="the targeted workspace can be named during creation"
+             value="8"/>
+      <entry name="activate"
+             summary="the targeted workspace can be created by activating it"
+             value="16"/>
+      <entry name="add_toplevel"
+             summary="the workspace toplevels can be created by adding a toplevel to it"
```

Comment:
This wording is a bit confusing because it's focused on creation, but the rest of this protocol reads like it can also be used for manipulating existing workspaces (e.g. name or activate exisiting workspaces, or move toplevels to existing workspaces).

Reply:
It is a bit confusing. Depending on the compositor, all of these things can create a workspace. E.g. Hyprland requires either activation or moving a toplevel, Niri requires activation(?), moving a toplevel, or assigning a name.

I'm not really sure how to make this clearer given the complexity of the problem.

Reply:
Potentially. I haven't touched on manipulation because its another can of worms, but reusing workspace targets to shuffle workspaces might make sense. Do you think that would need more than a create_from_other capability, which would effectively move the workspace to a new position?

Reply:
I think that should work for niri, though I'd call it something closer to "move"